File: src/main/java-graphql/io/github/graphql/model/CreatedRepositoryOrRestrictedContributionResponseProjection.java
package io.github.graphql.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField;
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection;

/**
 * Response projection for CreatedRepositoryOrRestrictedContribution
 */
@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:48+0800"
)
public class CreatedRepositoryOrRestrictedContributionResponseProjection extends GraphQLResponseProjection {

    public CreatedRepositoryOrRestrictedContributionResponseProjection() {
    }

    @Override
    public CreatedRepositoryOrRestrictedContributionResponseProjection all$() {
        return all$(3);
    }

    @Override
    public CreatedRepositoryOrRestrictedContributionResponseProjection all$(int maxDepth) {
        this.typename();
        return this;
    }

    public CreatedRepositoryOrRestrictedContributionResponseProjection onRestrictedContribution(RestrictedContributionResponseProjection subProjection) {
        return onRestrictedContribution(null, subProjection);
    }

    public CreatedRepositoryOrRestrictedContributionResponseProjection onRestrictedContribution(String alias, RestrictedContributionResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("...on RestrictedContribution").alias(alias).projection(subProjection));
        return this;
    }

    public CreatedRepositoryOrRestrictedContributionResponseProjection onCreatedRepositoryContribution(CreatedRepositoryContributionResponseProjection subProjection) {
        return onCreatedRepositoryContribution(null, subProjection);
    }

    public CreatedRepositoryOrRestrictedContributionResponseProjection onCreatedRepositoryContribution(String alias, CreatedRepositoryContributionResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("...on CreatedRepositoryContribution").alias(alias).projection(subProjection));
        return this;
    }

    public CreatedRepositoryOrRestrictedContributionResponseProjection typename() {
        return typename(null);
    }

    public CreatedRepositoryOrRestrictedContributionResponseProjection typename(String alias) {
        fields.add(new GraphQLResponseField("__typename").alias(alias));
        return this;
    }


}

File: src/main/scala-graphql/io/github/graphql/model/CreatedRepositoryOrRestrictedContributionResponseProjection.scala
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection

/**
 * Response projection for CreatedRepositoryOrRestrictedContribution
 */
@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-06T19:48:01+0800"
)
class CreatedRepositoryOrRestrictedContributionResponseProjection extends GraphQLResponseProjection {

    override def all$(): CreatedRepositoryOrRestrictedContributionResponseProjection = all$(3)

    override def all$(maxDepth: Int): CreatedRepositoryOrRestrictedContributionResponseProjection = {
        this.typename()
        this
    }

    def onRestrictedContribution(subProjection: RestrictedContributionResponseProjection): CreatedRepositoryOrRestrictedContributionResponseProjection = {
        onRestrictedContribution(null.asInstanceOf[String], subProjection)
    }

    def onRestrictedContribution(alias: String, subProjection: RestrictedContributionResponseProjection): CreatedRepositoryOrRestrictedContributionResponseProjection = {
        fields.add(new GraphQLResponseField("...on RestrictedContribution").alias(alias).projection(subProjection))
        this
    }

    def onCreatedRepositoryContribution(subProjection: CreatedRepositoryContributionResponseProjection): CreatedRepositoryOrRestrictedContributionResponseProjection = {
        onCreatedRepositoryContribution(null.asInstanceOf[String], subProjection)
    }

    def onCreatedRepositoryContribution(alias: String, subProjection: CreatedRepositoryContributionResponseProjection): CreatedRepositoryOrRestrictedContributionResponseProjection = {
        fields.add(new GraphQLResponseField("...on CreatedRepositoryContribution").alias(alias).projection(subProjection))
        this
    }

    def typename(): CreatedRepositoryOrRestrictedContributionResponseProjection = {
        typename(null.asInstanceOf[String])
    }

    def typename(alias: String): CreatedRepositoryOrRestrictedContributionResponseProjection = {
        fields.add(new GraphQLResponseField("__typename").alias(alias))
        this
    }


}
